prov/verbs/src/fi_verbs.c

```diff
@@ -1618,6 +1618,71 @@ static struct fi_ops_atomic fi_ibv_msg_ep_atomic_ops = {
 	.compwritevalid = fi_ibv_msg_ep_atomic_compwritevalid
 };
 
+/* RDMA CM doesn't tell us how big an address is, so we have to guess :( */
+static size_t fi_ibv_getaddrlen(struct sockaddr *sa)
+{
+	switch (sa->sa_family) {
+	case AF_INET:
+		return sizeof(struct sockaddr_in);
+		break;
+	case AF_INET6:
+		return sizeof(struct sockaddr_in6);
+		break;
+#ifdef AF_IB
+	case AF_IB:
+		return sizeof(struct sockaddr_ib);
+		break;
+#endif
+	default:
+		return sizeof(struct sockaddr);
+		break;
+	}
+}
+
+static int fi_ibv_copy_addr(void *dst_addr, size_t *dst_addrlen, void *src_addr)
+{
+	size_t src_addrlen;
+
+	src_addrlen = fi_ibv_getaddrlen(src_addr);
+	if (*dst_addrlen < src_addrlen) {
+		memcpy(dst_addr, src_addr, *dst_addrlen);
+		*dst_addrlen = src_addrlen;
+	} else {
+		memcpy(dst_addr, src_addr, src_addrlen);
+	}
```

Comment:
I think you need to return the correct size of the address in either case.  I.e. move the *dst_addrlen =   portion outside of the if -else clause